internal/extproc/chatcompletion_processor.go

```diff
@@ -0,0 +1,238 @@
+package extproc
+
+import (
+	"bytes"
+	"compress/gzip"
+	"context"
+	"fmt"
+	"io"
+	"log/slog"
+	"unicode/utf8"
+
+	corev3 "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	extprocv3 "github.com/envoyproxy/go-control-plane/envoy/service/ext_proc/v3"
+	"google.golang.org/protobuf/types/known/structpb"
+
+	"github.com/envoyproxy/ai-gateway/filterapi"
+	"github.com/envoyproxy/ai-gateway/internal/extproc/translator"
+	"github.com/envoyproxy/ai-gateway/internal/llmcostcel"
+)
+
+// NewChatCompletionProcessor creates a new processor for the chat completino requests
+func NewChatCompletionProcessor(config *processorConfig, logger *slog.Logger) ProcessorIface {
```

Comment:
This is the `Processor` object that I've renamed to ChatCompletionProcessor as now we can have different processors per request path. The contents are just a verbatim copy (modulo the type name change).